### PR TITLE
Spreadsheet: Add Range(s).first() and min(If)/max(If) and sumProduct(If) functions

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

**Review by alimpfard — COMMENTED** [review]
Please also add some tests when adding new stuff to this, otherwise the tests would be pretty much useless.

File: Base/res/js/Spreadsheet/runtime.js [hunk 1]
```diff
@@ -375,6 +375,20 @@ function min(cells) {
     return minIf(() => true, cells);
 }
 
+function sumProductIf(condition, rangeOne, rangeTwo) {
+    const rangeOneNums = numericResolve(rangeOne);
+    const rangeTwoNums = numericResolve(rangeTwo);
+    return rangeOneNums.reduce((prev, curr, i) => {
+        const prod = curr * rangeTwoNums[i];
+        if (!condition(curr, rangeTwoNums[i], prod)) return prev;
+        else return prev + prod;
```
[the remark this answers is not in the document]
Comment (u9g) [reply Base/res/js/Spreadsheet/runtime.js]: fixed :)

File: Base/res/js/Spreadsheet/runtime.js [hunk 2]
```diff
@@ -713,6 +729,48 @@ averageIf.__documentation = JSON.stringify({
...
+        "maxIf(x => x > 4, R`A1:C4`)":
+            "Finds the largest number within A1:C4 that is greater than 4",
+    },
+});
+
+min.__documentation = JSON.stringify({
+    name: "min",
+    argc: 1,
+    argnames: ["cell names"],
+    doc: "Finds the cell with the smallest value",
```
[the remark this answers is not in the document]
Comment (u9g) [reply Base/res/js/Spreadsheet/runtime.js]: fixed

File: Base/res/js/Spreadsheet/runtime.js [hunk 3]
```diff
@@ -713,6 +729,48 @@ averageIf.__documentation = JSON.stringify({
     },
 });
 
+max.__documentation = JSON.stringify({
+    name: "max",
+    argc: 1,
+    argnames: ["cell names"],
+    doc: "Finds the cell with the largest value",
```
[the remark this answers is not in the document]
Comment (u9g) [reply Base/res/js/Spreadsheet/runtime.js]: fixed

**Comment by u9g** [comment]
Added tests :^)

**Comment by alimpfard** [comment]
LGTM, let's just see it pass on CI.

**Pull request merged**